python/lsst/analysis/tools/actions/vector/calcRhoStatistics.py

```diff
@@ -0,0 +1,649 @@
+    the default values and excludes the key-values pairs when the item is None.
+
+    Notes
+    -----
+    This is intended to be used in CalcRhoStatistics class.
+    """
+
+    nbins = Field[int](
+        doc=(
+            "How many bins to use. "
+            "(Exactly three of nbins, bin_size, min_sep, max_sep "
+            "are required. If nbins is not given, it will be "
+            "calculated from the values of the other three, "
+            "rounding up to the next highest integer. "
+            "In this case, bin_size will be readjusted to account "
+            "for this rounding up."
+        ),
+        optional=True,
+        check=lambda x: x > 0,
+    )
+
+    bin_size = Field[float](
+        doc=(
+            "The width of the bins in log(separation). "
+            "Exactly three of nbins, bin_size, min_sep, max_sep are required. "
+            "If bin_size is not given, it will be calculated from the values "
+            "of the other three."
+        ),
+        optional=True,
+    )
+
+    min_sep = Field[float](
+        doc=(
+            "The minimum separation in units of sep_units, if relevant. "
+            "Exactly three of nbins, bin_size, min_sep, max_sep are required. "
+            "If min_sep is not given, it will be calculated from the values "
+            "of the other three."
+        ),
+        optional=True,
+    )
+
+    max_sep = Field[float](
+        doc=(
+            "The maximum separation in units of sep_units, if relevant. "
+            "Exactly three of nbins, bin_size, min_sep, max_sep are required. "
+            "If max_sep is not given, it will be calculated from the values "
+            "of the other three."
+        ),
+        optional=True,
+    )
+
+    sep_units = ChoiceField[str](
+        doc=(
+            "The units to use for the separation values, given as a string. "
+            "This includes both min_sep and max_sep above, as well as the "
+            "units of the output distance values."
+        ),
+        default="radian",
+        optional=True,
+        allowed={units: units for units in ["arcsec", "arcmin", "degree", "hour", "radian"]},
+    )
+
+    bin_slop = Field[float](
+        doc=(
+            "How much slop to allow in the placement of pairs in the bins. "
+            "If bin_slop = 1, then the bin into which a particular pair is "
+            "placed may be incorrect by at most 1.0 bin widths. "
+            r"If None, use a bin_slop that gives a maximum error of 10% on "
+            "any bin, which has been found to yield good results for most "
+            "applications."
+        ),
+        default=None,
+        optional=True,
+    )
+
+    brute = Field[bool](
+        doc=(
+            "Whether to use the 'brute force' algorithm? "
+            "Unlike treecorr, setting this to 1 or 2 is not supported."
+        ),
+        default=False,
+        optional=True,
+    )
```

Comment:
Pretty sure you wouldn't ever want this for rho stats...

Reply:
In fact, a bunch of these are irrelevant for this use case, so should probably be removed here.  Everything from here down to num_bootstrap I don't think should ever be changed from their default values.  (Many of them aren't even relevant for rho stats, so wouldn't do anything if you did change them.  For the others, it's just a bad idea.)

Reply:
I agree that `num_threads` should probably be removed, but this class is meant to map all the keywords that can be passed to instantiate any `BinnedCorr2` instances more generally. I don't have a use case beyond rho statistics for now, so I could go either way.